ckanext/validation/logic.py

```diff
@@ -97,40 +118,20 @@ def resource_validation_run(context, data_dict):
     # Check if there was an existing validation for the resource
     try:
         session = context['model'].Session
-        ValidationStatusHelper().createValidationJob(session, data_dict['resource_id'])
+        ValidationStatusHelper().createValidationJob(session, resource_id)
     except ValidationJobAlreadyEnqueued:
-        log.error("resource_validation_run: ValidationJobAlreadyEnqueued %s", data_dict['resource_id'])
-        return
+        if async_job:
+            log.error("resource_validation_run: ValidationJobAlreadyEnqueued %s", data_dict['resource_id'])
+            return
 
     if async_job:
         package_id = resource['package_id']
         enqueue_validation_job(package_id, resource_id)
     else:
-        # run_validation_job(resource_id)  # Plan is to only pass resource_id, but tests need to be fixed for this
+        # run_validation_job(resource_id)  # TODO only pass resource_id, tests need to be fixed for this
         run_validation_job(resource)
```

Comment:
could remove this todo, since inline job should use what it already has in memory (its not going onto redis queue etc)